### ISSUE TITLE
PassThroughTextMapPropagator

### ISSUE THREAD

**Issue by anuraaga** [open-telemetry/opentelemetry-specification#1717]
Some services do not need to participate in a trace but do need to propagate incoming context through to outgoing nodes. They can currently configure propagators such as `W3CTraceContextPropagator` and skip setting up an SDK, but there is unnecessary overhead in deserializing and serializing the exact same string. A pass-through propagator that accepts keys to propagate as Strings would be better for such cases.

**Comment by carlosalberto** [comment]
If I understand correctly, this means you would get the incoming context, keep it 'raw' and then simply propagate this text block?

A prototype for this would be interesting, along with a benchmark of how much performance improvement we would get (and definitely useful as I've lately been hearing some users asking for this).

**Comment by anuraaga** [comment]
Thanks @carlosalberto - sent a prototype for Java in incubator

https://github.com/open-telemetry/opentelemetry-java/pull/3256

**Comment by ahayworth** [comment]
Hi @anuraaga - we were discussing this as part of the Ruby SIG meeting today. Could you clarify why and when this would be useful? We were unable to come up with use-cases for such a thing, that couldn't be answered by "just pass through headers unmodified at the application level" (or something similar).

**Comment by carlosalberto** [comment]
FWIW @ahayworth you don't have to implement this (yet) to do your release ;)

**Comment by ahayworth** [comment]
Yep, we're just generally trying to stay abreast of changes, hence why we asked for more examples :)

**Comment by lacarvalho91** [comment]
@ahayworth not sure if this helps but we use the `PassThroughPropagator` because we have headers that need to be propagated for routing purposes (mostly for A/B tests)

using the `PassThroughPropagator` is nice because it allows us to leverage the mechanism already in place from OpenTelemetry for context propagation. So whilst we would just be able to write code ourselves that passes these unmodified headers down, its nice to be able to just leverage the mechanism already in place when using OTel.

also worth mentioning: we had some trouble getting in working with the auto configured OTel instance that is constructed by the Java agent. we've started a discussion for that [here](https://github.com/open-telemetry/opentelemetry-java-instrumentation/discussions/9489).

**Comment by mtwo** [comment]
Triage feedback: we've observed pass through propagators in several of our SDKs, however they're not in the specification / not required across all languages. Given the age of this issue and the potential impact on SDKs (they may enter states where the context is NULL), we'd like more feedback from the community about whether this is still needed.
